rpcs3/Emu/Cell/Modules/cellCamera.cpp

```diff
@@ -332,20 +458,40 @@ s32 cellCameraOpenEx(s32 dev_num, vm::ptr<CellCameraInfoEx> info)
 		return CELL_CAMERA_ERROR_NOT_INIT;
 	}
 
-	s32 read_mode = info->read_mode;
-
-	u32 status = set_and_send_read_mode(dev_num, read_mode);
+	s32 status = set_and_send_read_mode(dev_num, info->read_mode);
 	if (status != CELL_OK)
 	{
 		return status;
 	}
 
-	status = cellCameraSetAttribute(dev_num, CELL_CAMERA_GAMEPID, status, 0); // yup, that's what libGem does
+	status = cellCameraSetAttribute(dev_num, CELL_CAMERA_GAMEPID, 0, 0); // yup, that's what libGem does
 	if (status != CELL_OK)
 	{
 		return status;
 	}
 
+	if (!check_dev_num)
+	{
+		return CELL_CAMERA_ERROR_PARAM;
+	}
+
+	{
+		semaphore_lock lock(g_camera->mutex);
```

Comment:
Theres another semaphore_lock later for the same mutex in the function, this kinda seems redundant, is there a chance for refactor or just take the lock for the rest of the function?